source/extensions/quic_listeners/quiche/codec_impl.h

```diff
@@ -47,10 +47,7 @@ class QuicHttpServerConnectionImpl : public QuicHttpConnectionImplBase,
 
   // Http::Connection
   void goAway() override;
-  void shutdownNotice() override {
-    // TODO(danzh): Add double-GOAWAY support in QUIC.
```

Comment:
Can you leave the TODO there? It's not implemented in QUICHE yet. Or is it already implemented somehow in Envoy HCM? And shutdownNotice() is just part of the work flow?

Reply:
HCM [already calls `shutdownNotice()`](https://github.com/envoyproxy/envoy/blob/60bc17b301a03b097b270886d5b6746bd49a468b/source/common/http/conn_manager_impl.cc#L1120), which is used to send a first GOAWAY for H2 but it wasn't implemented by the quiche codec. This PR plugs the support I [added in quiche](https://quiche.googlesource.com/quiche/+/63eb43a60da8ad5d669891c553b22d4c14f88565) into Envoy to get the equivalent behavior for H3.

Reply:
ACK. Thanks for doing this!